simple_parsing/utils.py

```diff
@@ -479,6 +475,24 @@ def is_optional(t: Type) -> bool:
     return is_union(t) and type(None) in get_type_arguments(t)
 
 
+# Note: the typing module is still under development
+# and the way to detect Final and Literal may change in the future
+
+
+def is_final(t: Type) -> bool:
+    if sys.version_info < (3, 8, 0):
+        return False
```

Comment:
This isn't quite correct! People can use Final and Literal by importing them from the `typing_extensions` package, when their python versions are lower than 3.8.

I think adding something like this at the top of the file might be useful (btw, I wasn't aware of `typing_extensions` when I first wrote this module):
```python
try:
    from typing import Literal, Final, TypedDict # and every function you might need like get_origin etc
except ImportError:
    from typing_extensions import Iteral, Final, TypedDict, # and every function you might need
```

I'm not 100% sure, but I think the `typing_extensions` package might already have all the functions that `typing_inspect` provides.

Reply:
Yeah, I didn't know if we wanted to bring in a dependency on typing extensions. I can add it if you want.

Reply:
It's probably already installed because of typing_inspect I think

Reply:
There should also be a test for `is_final` and another for  `is_literal`. Something like this:
```python
@pytest.mark.parametrize("some_type, expected",
    [
        (Final, True),
        (Dict, False),
        (Final[str], True),
        (Tuple[str], False),
        ... # etc etc
    ]
)
def test_is_final(some_type: Type, expected: bool):
     assert is_final(some_type) is expected
```